src/RestSharp/RestClient.Async.cs

```diff
@@ -138,23 +153,65 @@ public partial class RestClient {
                     location = new Uri(url, location);
                 }
 
+                // Mirror HttpClient redirection behavior as of 07/25/2023:
+                // Per https://tools.ietf.org/html/rfc7231#section-7.1.2, a redirect location without a
+                // fragment should inherit the fragment from the original URI.
+                string requestFragment = originalUrl.Fragment;
+                if (!string.IsNullOrEmpty(requestFragment)) {
+                    string redirectFragment = location.Fragment;
+                    if (string.IsNullOrEmpty(redirectFragment)) {
+                        location = new UriBuilder(location) { Fragment = requestFragment }.Uri;
+                    }
+                }
+
+                // Disallow automatic redirection from secure to non-secure schemes
```

Comment:
That's one thing I want to be allowed based on redirect options.

Reply:
I'm happy to help do the work if you point me in the general direction of an API surface outline. :)

Reply:
Check my comment in #2059

Reply:
This specific decision is now driven by the new option object.